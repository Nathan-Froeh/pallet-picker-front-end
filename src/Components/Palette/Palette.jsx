import React, { Component } from 'react';
import './Palette.scss'

export class Palette extends Component {
  constructor(props) {
    super(props)
  }
  render() {
    return (
      <div className='project-palette'>
        <h4 className='tileTitle'>{this.props.name}</h4>
        <p className='tile' style={{backgroundColor: `#${this.props.color_1}`}}>{this.props.color_1}</p>
        <p className='tile' style={{backgroundColor: `#${this.props.color_2}`}}>{this.props.color_2}</p>        
        <p className='tile' style={{backgroundColor: `#${this.props.color_3}`}}>{this.props.color_3}</p>        
        <p className='tile' style={{backgroundColor: `#${this.props.color_4}`}}>{this.props.color_4}</p>        
        <p className='tile' style={{backgroundColor: `#${this.props.color_5}`}}>{this.props.color_5}</p>
      </div>
    )
  }
}

export default Palette
