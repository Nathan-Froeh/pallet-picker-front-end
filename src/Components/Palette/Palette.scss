.tile {
  // background-color: black;
  height: 20px;
  margin: 5px;
  width: 20px;
}