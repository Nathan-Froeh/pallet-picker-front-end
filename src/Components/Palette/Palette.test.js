import React from 'react';
import { shallow } from 'enzyme';
import Palette from './Palette';

describe('Palette', () => {
  let wrapper, instance;

  const props = {
    deletePalette: jest.fn(),
  }

  beforeEach(() => {
    wrapper = shallow (<Palette {...props}/>)
    instance = wrapper.instance()
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  }); 

  it('should call deletePalette from state when handleDelete is called', () => {
    wrapper.instance().handleDelete({preventDefault: jest.fn()});

  })

})