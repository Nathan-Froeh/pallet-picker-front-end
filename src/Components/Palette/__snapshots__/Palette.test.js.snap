// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette should match snapshot 1`] = `
<Popup
  arrow={true}
  arrowStyle={Object {}}
  className=""
  closeOnDocumentClick={true}
  closeOnEscape={true}
  contentStyle={Object {}}
  defaultOpen={false}
  disabled={false}
  keepTooltipInside={false}
  lockScroll={false}
  modal={true}
  mouseEnterDelay={100}
  mouseLeaveDelay={100}
  offsetX={0}
  offsetY={0}
  on={
    Array [
      "click",
    ]
  }
  onClose={[Function]}
  onOpen={[Function]}
  open={false}
  overlayStyle={Object {}}
  position="bottom center"
  repositionOnResize={true}
  trigger={
    <div
      className="mini-palette"
      onClick={[Function]}
    >
      <h4
        className="tileTitle"
      />
      <p
        className="tile"
        style={
          Object {
            "backgroundColor": "undefined",
          }
        }
      />
      <p
        className="tile"
        style={
          Object {
            "backgroundColor": "undefined",
          }
        }
      />
      <p
        className="tile"
        style={
          Object {
            "backgroundColor": "undefined",
          }
        }
      />
      <p
        className="tile"
        style={
          Object {
            "backgroundColor": "undefined",
          }
        }
      />
      <p
        className="tile"
        style={
          Object {
            "backgroundColor": "undefined",
          }
        }
      />
    </div>
  }
>
  <div
    className="project-palette"
    onClick={[Function]}
  >
    <h4
      className="tileTitle"
    />
    <section>
      <div
        className="color-pad"
        style={
          Object {
            "backgroundColor": "undefined",
          }
        }
      >
        <p />
      </div>
      <div
        className="color-pad"
        style={
          Object {
            "backgroundColor": "undefined",
          }
        }
      >
        <p />
      </div>
      <div
        className="color-pad"
        style={
          Object {
            "backgroundColor": "undefined",
          }
        }
      >
        <p />
      </div>
      <div
        className="color-pad"
        style={
          Object {
            "backgroundColor": "undefined",
          }
        }
      >
        <p />
      </div>
      <div
        className="color-pad"
        style={
          Object {
            "backgroundColor": "undefined",
          }
        }
      >
        <p />
      </div>
    </section>
    <button
      onClick={[Function]}
    >
      Delete Palette
    </button>
  </div>
</Popup>
`;
